src/julia_gc.c

```diff
@@ -685,7 +701,7 @@ static void PostGCHook(int full)
 // is called by the Julia GC whenever it marks a GAP master pointer object)
 static uintptr_t JMarkMPtr(jl_ptls_t ptls, jl_value_t * obj)
 {
-    if (!*(void **)obj)
+    if (!*(void **)obj || !jl_typeis(obj, datatype_mptr))
```

Comment:
But `obj` does not point at the start of the mptr, `BAG_HEADER((Bag)obj)` does.

Reply:
Yes, that was actually an unnecessary test, too.